### PR TITLE
feat: Add `X-Gu-CDK-Version` tag to GuStack

### PR DESCRIPTION
**Pull request by akash1810**
## What does this change?

Adds `X-Gu-CDK-Version` tag to `GuStack`. We can use this to track which stacks have been created using the library, and with which version. Adding the tag to `GuStack` means it'll get applied to all resources.

The absence of this tag suggests the stack hasn't been created with the cdk library. Using the version number allows us to track how up to date stacks are too.

We mock the library version in tests to remove having to update snapshots whenever we publish a new version of the library, which, among other things, would make automated publishing difficult! The mock is applied globally for simplicity.

## Does this change require changes to existing projects or CDK CLI?

Yes! The change is small though, it should be a matter of bumping the version of this library being used.

## How to test

Observe existing tests.

## How can we measure success?

We know what version of this library was used to create a stack, if any.

## Have we considered potential risks?

~The version number is retrieved from the `package.json`. Does this resolve the the correct value once published? That is, if we use the library in a project, is the version number that of guardian/cdk or of the consuming project?~ Update: this is fine. The import path is relative so will always resolve to this package's `package.json` file.

### CODE REVIEW

File: src/constants/__mocks__/library-info.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export const LibraryInfo = {
```
Comment (akash1810) [inline src/constants/__mocks__/library-info.ts:1]: see https://jestjs.io/docs/en/manual-mocks

File: jest.config.js [hunk 2]
```diff
@@ -3,4 +3,5 @@ module.exports = {
   transform: {
     "^.+\\.tsx?$": "ts-jest",
   },
+  setupFilesAfterEnv: ["./jest.setup.js"],
```
Comment (akash1810) [inline jest.config.js:6]: see https://jestjs.io/docs/en/configuration.html#setupfilesafterenv-array

**Comment by jacobwinch** [comment]
> We know what version of this library was used to create a stack, if any.

Great idea 💡 I think this will be really useful in the future, especially if we can easily find stacks using specific versions via Prism!

**Review by jacobwinch — APPROVED** [review]
Great work 💯 

As far as I can tell, the only downside of this approach is that it will attempt to modify a large number of resources whenever users upgrade the library (because we'll need to perform tag updates, even if the underlying resources are unchanged)? 

I think it'll be really useful to have this info, but perhaps we should document this behaviour so that users are not concerned by these snapshot updates or change sets?

**Comment by akash1810** [comment]
> As far as I can tell, the only downside of this approach is that it will attempt to modify a large number of resources whenever users upgrade the library (because we'll need to perform tag updates, even if the underlying resources are unchanged)?

Ah great point! Will have a look to see if a tag can be applied only to the stack and not spread across all its resources.

**Comment by akash1810** [comment]
> Will have a look to see if a tag can be applied only to the stack and not spread across all its resources.

From the [docs](https://docs.aws.amazon.com/cdk/latest/guide/tagging.html#tagging_props) it is possible to choose which resources to add a tag to, however it looks like tags for the stack can only be set at [create/update time](https://docs.aws.amazon.com/cli/latest/reference/cloudformation/create-stack.html#id2:~:text=%2D%2Dtags%20(list))*, that is, they cannot be part of the template.

There is [this hack(?)](https://stackoverflow.com/a/37025655/3868241), but I'm not particularly keen in it as it is pretty obscure and unintuitive.

I see three options to progress:
1. Document this behaviour and merge
2. Make the tag more static, something like `X-Gu-Stack-Author => guardian/cdk`
3. Make it easier to apply updates through CD, which can hide the verbose change sets away

\* as we can't automate the application of tags against the stack, our tracking strategy might need to change.

**Review by jacobwinch — APPROVED** [review]
> I think I favour 1 ([3245a2c](https://github.com/guardian/cdk/commit/3245a2c174527551c91f3032968cc0e74ded2c86)).

Yep, I agree - thanks for looking into alternatives though! 

👍 🚢

**Pull request merged**